app/src/main/java/com/esri/arcgisruntime/opensourceapps/datacollection/views/popup/PopupView.kt

```diff
@@ -168,33 +180,152 @@ class PopupView : FrameLayout {
+                        TODO("Not yet implemented")
+                    }
+
+                    override fun onItemSelected(
+                        parent: AdapterView<*>?,
+                        view: View?,
+                        position: Int,
+                        id: Long
+                    ) {
+                        popupManager.updateValue(
+                            codedValueDomain.codedValues[position].code,
+                            popupField
+                        )
+                    }
+                }
+        }
+
+        /**
+         * Updates the value of the specified PopupField to the appropriately cast string value of
+         * the specified value
+         */
+        private fun updateValue(popupField: PopupField, newValue: String): ArcGISRuntimeException? {
+            var error: ArcGISRuntimeException? = null
+            when (popupManager.getFieldType(popupField)) {
+                Field.Type.SHORT -> error =
+                    if (newValue.toShortOrNull() != null) {
+                        popupManager.updateValue(newValue.toShort(), popupField)
+                    } else {
+                        popupManager.updateValue(newValue, popupField)
+                    }
+                Field.Type.INTEGER -> error =
+                    if (newValue.toIntOrNull() != null) {
+                        popupManager.updateValue(newValue.toInt(), popupField)
+                    } else {
+                        popupManager.updateValue(newValue, popupField)
+                    }
+                Field.Type.FLOAT -> error =
+                    if (newValue.toFloatOrNull() != null) {
+                        popupManager.updateValue(newValue.toFloat(), popupField)
+                    } else {
+                        popupManager.updateValue(newValue, popupField)
+                    }
+                Field.Type.DOUBLE -> error =
+                    if (newValue.toDoubleOrNull() != null) {
+                        popupManager.updateValue(newValue.toDouble(), popupField)
+                    } else {
+                        popupManager.updateValue(newValue, popupField)
+                    }
+                Field.Type.TEXT -> error = popupManager.updateValue(newValue, popupField)
+
+                else -> Log.i(
+                    "PopupView",
+                    "Unhandled field type: " + popupManager.getFieldType(popupField)
+                )
+            }
+            return error
+        }
+
+        /**
+         * Returns the position of the given code in list of coded values. If no match found,
+         * then the first value in the list is returned.
+         *
+         * @param code the code whose position is to be determined in the coded values list
+         * @param codedValues the list of coded values
+         * */
+        private fun getCodedValuePosition(code: Any, codedValues: List<CodedValue>): Int {
```

Comment:
I think this function can be replaced with `codedValues.indexOfFirst { it.code == code }`